agent/bench-scripts/pbench-fio

```diff
@@ -629,40 +630,40 @@ function fio_run_job() {
 
 	echo "running fio job: $fio_job_file ($(basename $benchmark_results_dir))"
 
-	if [ ! -z "$clients" ]; then
+	if [ ! -z "${clients}" ]; then
 		debug_log "creating directories on the clients"
 		for client in "${client_names[@]}"; do
-			ssh $ssh_opts $client mkdir -p $benchmark_results_dir &
+			ssh ${ssh_opts} ${client} mkdir -p ${benchmark_results_dir} &
 		done
 		wait
-		debug_log "opening port 8765 on firewall on the clients"
+		debug_log "opening client port in firewall on the clients"
 		for client in "${client_names[@]}"; do
-			ssh $ssh_opts $client "firewall-cmd --add-port=${client_ports[$client]:-$fio_server_port}/tcp >/dev/null" &
+			ssh ${ssh_opts} ${client} "firewall-cmd --add-port=${client_ports[${client}]}/tcp >/dev/null" &
 		done
 		wait
 		debug_log "killing any old fio process on the clients"
 		for client in "${client_names[@]}"; do
-			ssh $ssh_opts $client "killall fio >/dev/null 2>&1" &
+			ssh ${ssh_opts} ${client} "killall fio >/dev/null 2>&1" &
 		done
 		wait
 		debug_log "starting new fio process on the clients"
 		for client in "${client_names[@]}"; do
-			ssh $ssh_opts $client "pushd $benchmark_results_dir >/dev/null; screen -dmS fio-server bash -c ''$benchmark_bin' --server${client_ports[$client]:+=,${client_ports[$client]}} 2>&1 >client-result.txt'"
+			ssh ${ssh_opts} ${client} "pushd ${benchmark_results_dir} >/dev/null; screen -dmS fio-server bash -c ''$benchmark_bin' --server=,${client_ports[${client}]} 2>&1 >client-result.txt'"
 		done
 		wait
 		debug_log "waiting for fio process(server) to start on clients"
 		for client in "${client_names[@]}"; do
-			timeout --kill-after=1 60 bash -c 'until printf "" 2>>/dev/null >>/dev/tcp/$0/$1; do sleep 1; done' $client ${client_ports[$client]:-${fio_server_port}}
+			timeout --kill-after=1 60 bash -c 'until printf "" 2>>/dev/null >>/dev/tcp/$0/$1; do sleep 1; done' "${client}" "${client_ports[${client}]}"
```

Comment:
Very simply, if you don't enclose those variable expansions in quotes, then you end up passing potentially more than 2 arguments.  The quotes ensure we pass 2 arguments.

As for quoting internally, from past experience debugging when these shell scripts fail due to variable expansion issues, it is easier to track down when you can match argument counts provided to those expected.  Then when an error occurs, we know where to look.

In this case no matter what we do, we'll have a problem.  With them enclosed in double quotes, the `$0` and `$1` inner references will expand with the proper values, but the `bash -c` script will fail entirely, where it might not if the resulting variables in the `bash -c` script happen to work.